core/deployment/src/main/java/io/quarkus/deployment/configuration/BuildTimeConfigurationReader.java

```diff
@@ -683,16 +683,16 @@ public void accept(final ConfigSource buildSystem) {
                 if (buildTimeNames.contains(name)) {
                     unknownBuildProperties.remove(property);
                     ConfigValue value = config.getConfigValue(property);
-                    if (value.getRawValue() != null) {
-                        allBuildTimeValues.put(value.getNameProfiled(), value.noProblems().withValue(value.getRawValue()));
+                    if (value.getValue() != null) {
+                        allBuildTimeValues.put(value.getNameProfiled(), value);
```

Comment:
Didn't we do that to fix another issue? I have a vague recollection of reviewing a PR with this change.